casanovo/data/ms_io.py

```
@@ -185,33 +254,44 @@ def save(self) -> None:
-                        "null",  # end
-                        # opt_global_aa_scores
-                        ",".join(list(map("{:.5f}".format, psm.aa_scores))),
-                        psm.sequence,  # opt_global_cv_MS:1003169_proforma_peptidoform_sequence
-                    ]
-                )
+                run_idx = self._run_map[filename]
+                scan_col = "null"
+                if Path(filename).suffix.lower() == ".mgf":
+                    # Normalize idx to "index=N" format, handling both
+                    # bare numeric IDs ("0") and prefixed IDs ("index=0").
+                    if idx.isnumeric():
```

Ruff (0.15.7)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)